grpc/federation/cel/list_test.go

```diff
@@ -64,11 +68,271 @@ func TestList(t *testing.T) {
+		{
+			name: "sortDesc field selection",
+			expr: `[
+grpc.federation.cel.test.Message{id: "a"}, 
+grpc.federation.cel.test.Message{id: "c"}, 
+grpc.federation.cel.test.Message{id: "b"}
+].sortDesc(v, v.id)`,
+			cmp: func(got any) error {
+				lister, ok := got.(traits.Lister)
+				if !ok {
+					return fmt.Errorf("invalid result type: %T", got)
+				}
+				expected := []*testpb.Message{
+					{
+						Id: "c",
+					},
+					{
+						Id: "b",
+					},
+					{
+						Id: "a",
+					},
+				}
+				if lister.Size().(types.Int) != types.Int(len(expected)) {
+					return fmt.Errorf("invalid size")
+				}
+
+				gotV, err := lister.ConvertToNative(reflect.TypeOf([]*testpb.Message{}))
+				if err != nil {
+					return fmt.Errorf("failed to conver to native: %w", err)
+				}
+				if diff := cmp.Diff(gotV, expected, cmpopts.IgnoreUnexported(testpb.Message{})); diff != "" {
+					return fmt.Errorf("(-got, +want)\n%s", diff)
+				}
+				return nil
+			},
+		},
+		{
+			name: "sortStableAsc",
+			expr: `[
+grpc.federation.cel.test.Message{id:"A", num:25}, 
+grpc.federation.cel.test.Message{id:"A", num:75}, 
+grpc.federation.cel.test.Message{id:"A", num:75}, 
+grpc.federation.cel.test.Message{id:"B", num:75},
+grpc.federation.cel.test.Message{id:"B", num:25},
+grpc.federation.cel.test.Message{id:"C", num:25},
+grpc.federation.cel.test.Message{id:"E", num:75},
+grpc.federation.cel.test.Message{id:"E", num:25},
+].sortStableAsc(v, v.num)`,
+			cmp: func(got any) error {
+				lister, ok := got.(traits.Lister)
+				if !ok {
+					return fmt.Errorf("invalid result type: %T", got)
+				}
+				expected := []*testpb.Message{
+					{
+						Id:  "A",
+						Num: 25,
+					},
+					{
+						Id:  "B",
+						Num: 25,
+					},
+					{
+						Id:  "C",
+						Num: 25,
+					},
+					{
+						Id:  "E",
+						Num: 25,
+					},
+					{
+						Id:  "A",
+						Num: 75,
+					},
+					{
+						Id:  "A",
+						Num: 75,
+					},
+					{
+						Id:  "B",
+						Num: 75,
+					},
+					{
+						Id:  "E",
+						Num: 75,
+					},
+				}
+				if lister.Size().(types.Int) != types.Int(len(expected)) {
+					return fmt.Errorf("invalid size")
+				}
+
+				gotV, err := lister.ConvertToNative(reflect.TypeOf([]*testpb.Message{}))
+				if err != nil {
+					return fmt.Errorf("failed to conver to native: %w", err)
+				}
+				if diff := cmp.Diff(gotV, expected, cmpopts.IgnoreUnexported(testpb.Message{})); diff != "" {
+					return fmt.Errorf("(-got, +want)\n%s", diff)
+				}
+				return nil
+			},
+		},
+		{
+			name: "sortStableDesc",
+			expr: `[
+grpc.federation.cel.test.Message{id:"A", num:25}, 
+grpc.federation.cel.test.Message{id:"A", num:75}, 
+grpc.federation.cel.test.Message{id:"A", num:75}, 
+grpc.federation.cel.test.Message{id:"B", num:75},
+grpc.federation.cel.test.Message{id:"B", num:25},
+grpc.federation.cel.test.Message{id:"C", num:25},
+grpc.federation.cel.test.Message{id:"E", num:75},
+grpc.federation.cel.test.Message{id:"E", num:25},
+].sortStableDesc(v, v.num)`,
```

Comment:
If you're testing `sortStableDesc`, I think it's better to run it twice and verify that the ids are also in descending order.

`[].sortStableDesc(v, v.num).sortStableDesc(v, v.id)`